lib/super_digit.rb

```diff
@@ -1,15 +1,49 @@
+require 'benchmark'
+
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n), it depends on how many digits of a number, so it's O(n). 
+# Space Complexity - O(1), only use two extra variables, ie sum and result, to do the calculation. So the Space complexity is a constant.
 def super_digit(n)
```

Comment:
This has O(d^2) time complexity where d is the number of digits and because of the recursive method O(d) space complexity.